Simple Makefile to build extensions.

boumenot opened the pull request:
Use the version and type (name) from an extension's manifest.xml to
create an extension's zip archive.

Support building with directories that contain whitespace.

ahmetalpbalkan commented:
while you have the chance, let's switch to a little bit more strict bash mode http://redsymbol.net/articles/unofficial-bash-strict-mode/

ahmetalpbalkan commented on Makefile:
```diff
@@ -0,0 +1,20 @@
+$(EXTENSIONS):
+	$(eval NAME    = $(shell grep -Pom1 "(?<=<Type>)[^<]+" $@/manifest.xml))
+	$(eval VERSION = $(shell grep -Pom1 "(?<=<Version>)[^<]+" $@/manifest.xml))
+	@sh script/create_zip.sh $@ $(NAME) $(VERSION)
```
hm you shouldn't need `sh`?

boumenot replied:
Because the file isn't executable.  I can make that change instead.

ahmetalpbalkan commented on script/create_zip.sh:
```diff
@@ -37,7 +37,7 @@ ext_name=$2
 cd $ext_dir
 ext_dir=`pwd`
-cd $curr_dir
+cd "$curr_dir"
```
I find this `pwd` thing pretty sloppy. we can use relative paths to the build script by using `readonly SCRIPT_DIR=$(dirname $0)` and find our way from there

boumenot commented:
Slight change of plans.  There is considerably more work here than I plan on doing for this iteration.  The goal is to get a build that produces a deployable version of CustomScript.  To that end, I reverted create_zip.sh work because it is unnecessary.  I have changed the Makefile to do the work necessary to create a CustomScript deployable artifact.

There's more debt to pay off, but I'm deferring that for another day.

ahmetalpbalkan commented:
gotta invite someone with makefile literacy here

boumenot commented:
@jasonzio - would you please take a look?

jasonzio commented:
lgtm
